src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/MultiSourcePackageMetadataProvider.cs

```diff
@@ -114,20 +122,24 @@ public async Task<IPackageSearchMetadata> GetPackageMetadataAsync(PackageIdentit
             bool includeUnlisted,
             CancellationToken cancellationToken)
         {
+            cancellationToken.ThrowIfCancellationRequested();
             var tasks = _sourceRepositories
                 .Select(r => GetMetadataTaskSafeAsync(()=> r.GetPackageMetadataListAsync(packageId, includePrerelease, includeUnlisted, cancellationToken)))
                 .ToArray();
 
             var completed = (await Task.WhenAll(tasks))
                 .Where(m => m != null);
 
+            cancellationToken.ThrowIfCancellationRequested();
             var packages = completed.SelectMany(p => p);
 
+            cancellationToken.ThrowIfCancellationRequested();
```

Comment:
Zero work has been done here between the two calls to `ThrowIfCancellationRequested`. Remember, LINQ uses deferred execution.